engine/src/main/java/org/terasology/rendering/openvrprovider/OpenVRState.java

```diff
@@ -0,0 +1,100 @@
+package org.terasology.rendering.openvrprovider;
+
+import jopenvr.*;
+import org.joml.Matrix4f;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/* Contains all of the information that the user will need from OpenVR without using any OpenVR data structures. The
+OpenVRProvider automatically updates this.
+ */
+public class OpenVRState {
+    public static int leftEye = JOpenVRLibrary.EVREye.EVREye_Eye_Left;
+    public static int rightEye = JOpenVRLibrary.EVREye.EVREye_Eye_Right;
+
+    // Controllers
+    private static Matrix4f[] controllerPose = new Matrix4f[2];
+    private static VRControllerState_t[] lastControllerState = new VRControllerState_t[2];
+
+    private List<ControllerListener> controllerListeners = new ArrayList<>();
+
+    // In the head frame
+    private Matrix4f[] eyePoses = new Matrix4f[2];
+    private Matrix4f[] projectionMatrices = new Matrix4f[2];
+
+    // In the tracking system intertial frame
+    private Matrix4f headPose = new Matrix4f(1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1);
```

Comment:
My eyes are crossing with all these 1s and 0s. How about you add a method to MatrixUtils:

`public static Matrix4f createIdentityMatrix4f() {return new Matrix4f(1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1)}`